### DISCUSSION TITLE
Deepstream + Docker on Jetson Orin Nano

### DISCUSSION THREAD

**Discussion by jvd-monteiro** [ultralytics/ultralytics#15708] [General]
Hello,

I'm using a Jetson Orin Nano and wanted to try different set ups to run a YOLO model with Deepstream.
When setting things locally I can make it work, but when setting up a Docker with:

```
t=ultralytics/ultralytics:latest-jetson-jetpack6
sudo docker pull $t && sudo docker run -it --ipc=host --runtime=nvidia $t
```

I can make regular inference work, but was still not able to do it with Deepstream. So my main question is on how to set up a Container with Deepstream appropriately, since the one created above is not working... Should I create the NVIDIA Deepstream container itself and then set up Ultralytics libraries inside it? Please shed some light on the best approach for it.

Thanks in advance!

**Comment by glenn-jocher** [comment]
@jvd-monteiro hello,

For running YOLO with DeepStream in a Docker container on Jetson Orin Nano, it's recommended to start with an NVIDIA DeepStream container and then install the Ultralytics libraries within it. This ensures compatibility with DeepStream. You can follow the DeepStream setup guide at https://docs.ultralytics.com/guides/deepstream-nvidia-jetson/ for detailed instructions.

Thanks!

**Reply by jvd-monteiro** [reply]
Thats great @glenn-jocher, will try it.
Thanks for the quick response!

**Reply by glenn-jocher** [reply]
@jvd-monteiro you're welcome! If you encounter any issues or have further questions, feel free to ask.
